statsmodels/nonparametric/_smoothers_lowess.pyx

```diff
@@ -51,13 +57,19 @@ def lowess(np.ndarray[DTYPE_t, ndim = 1] endog,
     delta : float
         Distance within which to use linear-interpolation
         instead of weighted regression.
+    given_xvals : bool
+        Whether xvals was provided as a an argument or whether
+        we are just using xvals = exog
 
     Returns
     -------
     out : numpy array
         A numpy array with two columns. The first column
         is the sorted x values and the second column the
         associated estimated y-values.
+    resid_weights: numpy array
```

Comment:
Are there different from weights?  Also (throughout ideally, return type should be `ndarray` as in `resid_weights: ndarray`.

Reply:
'weights' are the data point weights for regressing at a specific point, while `resid_weights` is used for the overall down-weighting of outliers produced from multiple iterations and so don't depend upon the x values.